userspace/libsinsp/parsers.cpp

```diff
@@ -592,6 +594,57 @@ bool sinsp_parser::reset(sinsp_evt *evt)
 		evt->init();
 	}
 
+	// determine the event source
+	if (evt->m_info->category & EC_METAEVENT)
+	{
+		// metaevent are internal events, and as such they have no concrete event source
+		evt->m_source_idx = sinsp_no_event_source_idx;
+		evt->m_source_name = sinsp_no_event_source_name;
+	}
+	else if (evt->m_info->category & EC_PLUGIN)
+	{
+		// plugin events are produced by plugins of a certain ID, and have the
+		// event source specified by their producer plugin
+		if (evt->get_type() != PPME_PLUGINEVENT_E)
+		{
+			throw sinsp_exception("unknown plugin event type in sinsp parser: " + std::to_string(evt->get_type()));
+		}
+		bool pfound = false;
+		auto parinfo = evt->get_param(0);
+		ASSERT(parinfo->m_len == sizeof(int32_t));
+		uint32_t pgid = *(int32_t *) parinfo->m_val;
+		auto srcidx = m_inspector->get_plugin_manager()->source_idx_by_plugin_id(pgid, pfound);
+		if (!pfound)
+		{
+			evt->m_source_idx = sinsp_no_event_source_idx;
+			evt->m_source_name = sinsp_no_event_source_name;
+		}
+		else
+		{
+			evt->m_source_idx = srcidx;
+			evt->m_source_name = m_inspector->event_sources()[srcidx].c_str();
+		}
+	}
+	else
+	{
+		// every other event falls under the "syscall" event source umbrella
+		// cache index of "syscall" event source in case we haven't already
+		if (m_syscall_event_source_idx == sinsp_no_event_source_idx)
```

Comment:
I'd comment here stating that the `m_syscall_event_source_idx` is always the first (0) at the moment, but the code wants to be as independent as possible from this assumption.